### ISSUE TITLE
Mixing lighting effects?

### ISSUE THREAD

**Issue by Valachord** [polychromatic/polychromatic#280]
Is it possible to mix lighting effects like on Synapse? If not, is it a feature being worked on?

This application is amazing, I absolutely love it and have so much praise for the developer! It's almost as customizable as the Synapse app, this feature is actually the only thing I find missing!

Also, in the app, my keyboard shows up correctly as Hunstman Elite, name-wise, but the layout when customizing is that of the BlackWidow Stealth, which means that I can't customize the media buttons (their light is just turned off) or the underglow/lightbar that goes around the keyboard.
Am I supposed to do something to change that or has my keyboard just not been fully implemented in the application?

**Comment by lah7** [comment]
Thank you, I'm glad you love it! :smile: 

Mixing lighting effects as a layer system is planned for **v1.0.0**. I got a bit sidetracked when working on this feature (and things are taking a while) since I've re-written the Controller application away from being a hybrid "web" app, amongst other life stuff.

Interactive custom effects, like a "reactive" layer would come after v1.0.0, since it needs a bit more research to capture keys across different locales, layouts and without needing to be root.

----

Currently (v0.3.12) the keyboard graphic for creating custom lighting is hardcoded for a **BlackWidow Chroma**, so the current application is unaware of the matrix for different devices. [There are some pull requests patching the graphic](https://github.com/polychromatic/polychromatic/pulls?q=is%3Apr+is%3Aopen+label%3A%22device+specific%22), but the old codebase is frozen, and none for Hunstman.

It would be most helpful if you run this script and let me know what each key/position lights up for your **Hunstman Elite** - so once the "new effect editor" is ready, I can use the data to create a new graphic for your hardware:

https://raw.githubusercontent.com/polychromatic/polychromatic/stable-python38/tools/step-through-matrix.py

**Comment by Valachord** [comment]
Hi, I just came back to Linux/polychromatic after a long break, and I haven't seen your answer until now!
The app has certainly come a long way since I posted that, it's even better now, so damn awesome! :smile: :+1: 

Anyway, I can see that both my Huntsman elite and Basilisk Ultimate (both mouth and dock) are showing up properly now, and after spending a couple of minutes in the effects grid mode, I figured out what is what and have made some awesome layouts! Though, as far as I can tell, it seems that the option to chose profiles for specific applications has been removed? Is that right, or is it just me that needs glasses? :smile: 

I've also opened another issue with a [Controller Error](https://github.com/polychromatic/polychromatic/issues/321) I'm getting.

**Comment by lah7** [comment]
Glad to hear! 👍🏼 

>  it seems that the option to chose profiles for specific applications has been removed?

"Application Profiles" was a little ambiguous. They're "effects" now, but in future they could also be "presets" to have settings predetermined for both your keyboard and mouse. Essentially, effects can be both for aesthetics or for use under specific applications/games. No functionality lost. 🙂 

Coming from v0.3.12, old ones should have been converted. Choosing an icon still lists system applications, and game icons from Steam (if installed).

> EDIT: Forgot to ask, do you still need me to do that?

Only if you'd like a _pretty_ graphic for your hardware. The grid acts as a fallback.

There is a new graphic for **BlackWidow Elite**, which looks very similar to your **Huntsman Elite**. You could try that and see if they are compatible.

Otherwise, if you like, follow the WIP documentation here to contribute a graphic:
https://polychromatic.app/docs/devicemaps/
